### ISSUE TITLE
Getting lombok dependency missing error on non-java project

### ISSUE THREAD

**Issue by lexandro** [mplushnikov/lombok-intellij-plugin#263]
The recent version of the lombok plugin tries to add itself to every project, regardless of the project type. I'm working on a golang project, but every time, when I start intellij or open the project, I'm getting the following error:

```
11:51 AM    Lombok Dependency is missing
      Project "go-restful-middleware" does not seem to have lombok dependency added.
      This plugin does not provide it and will not function correctly without.
      Please make sure to add it manually and refresh your project.
      Click here to see how to add it to your project.
```

**Comment by mplushnikov** [comment]
You can disable plugin in the configuration for your project:
![image](https://cloud.githubusercontent.com/assets/1204509/17098725/4574b718-5264-11e6-95f7-7f633dd8490e.png)
But I will look for a chance to disable it automatically for non java projects by default.

**Comment by lexandro** [comment]
Cool. Thanks.

**Comment by relgames** [comment]
It also shows warnings in all projects not using lombok.
Did you consider adding a configuration option to turn warnings off completely?
I just like how it used to be before the update.

**Comment by AlexejK** [comment]
For projects where you are not using Lombok you can untick "Enable Lombok plugin for this project".

As a background - main reason for this check being added is due to the fact that many users of this plugin do not understand the fact that they need lombok and keep reporting issues, even though we are clearly stating that the Lombok dependency is needed.

I can understand that there is some frustration about some warnings and we can let users opt-out of this, just like with version check - but it's a per-project configuration.

**Comment by relgames** [comment]
> per-project configuration
I use lombok in maybe 1 project out of 10. Did you consider adding an option to disable all warnings altogether, not on a per-project basis?

**Comment by AlexejK** [comment]
Well if its a non-java project we can improve detection and not give warnings (ideally)
However configuration of the plugin is saved per project today, and we will have to investigate if something different could be done and when.

You can also change project defaults and set Lombok to disabled completely (even better, to avoid triggering augmentations for a "fast bailout" etc) or disable some features as default and only enable where you need on a per-project basis.

We do take developer experience seriously and will think if there are good way to change/improve this. For now, as a work-around at least, this is the best I can give you.

**Comment by AlexejK** [comment]
To mitigate this issue to some extend, I've made #266 that should allow you disable "missing lombok dependency" warning for the project that is non-java or where your project is properly configured but has non-java modules.

**Comment by AlexejK** [comment]
In addition, there has been discovered a bug with the new warnings - default setting was supposed to be disabled for some of the warnings (like out-of-date warning) but it was not. This is also fixed in #266.  See #267 bug for more details.

Default "state" for both "missing lombok" and "out-of-date lombok" should be disabled. So this should solve even bigger chunk of the issue :)

**Comment by stianlagstad** [comment]
I too find it annoying that the plugin gives me a warning in all projects where lombok is not used. Maybe you could add a "Don't show this again" option?

**Comment by AlexejK** [comment]
To sum it all up.

1. There is an issue with the release build where warnings got enabled automatically, even though default was ment to be "off".
2. A new configuration to disable lombok dependency (not only version) has been added
3. It might be a good idea to lead you into configuration for version check and what was mentioned in point 2 to disable warnings.

As the initial effort to reduce annoyance caused by the bug (point 1) and also make it a bit more flexible (point 2), the fix #266  has been merged. 
Point 3 is a good way forward when warnings are going to be enabled by default, but this is not a deal breaker given 1. and 2. are fixed.

**Comment by mplushnikov** [comment]
New version with mentioned fix #266  and some other improvements was released few minutes ago!

**Comment by OrangeDog** [comment]
>For projects where you are not using Lombok you can untick "Enable Lombok plugin for this project".

This doesn't help when only some of the modules in a project use Lombok.
